src/test/java/org/rrd4j/core/UtilTest.java

```diff
@@ -0,0 +1,261 @@
+        Assert.assertEquals(500, Util.max(500, -500), 0.0);
+        Assert.assertEquals(0, Util.max(-500, 0), 0.0);
+        Assert.assertEquals(-500, Util.max(-500, -1000), 0.0);
+    }
+
+    @Test
+    public void testMin() {
+        // Test with arrays of doubles
+        Assert.assertEquals(Double.NaN, Util.min(new double[] {}), 0.0);
+        Assert.assertEquals(0, Util.min(new double[] {Double.NaN, 0}), 0.0);
+        Assert.assertEquals(Double.POSITIVE_INFINITY, Util.min(new double[] {Double.NaN, Double.POSITIVE_INFINITY}), 0.0);
+        Assert.assertEquals(-500, Util.min(new double[] {500, -500}), 0.0);
+        Assert.assertEquals(-500, Util.min(new double[] {-500, 0}), 0.0);
+        Assert.assertEquals(-1000, Util.min(new double[] {-500, -1000}), 0.0);
+
+        // Test with pair of doubles
+        Assert.assertEquals(0, Util.min(Double.NaN, 0), 0.0);
+        Assert.assertEquals(Double.POSITIVE_INFINITY, Util.min(Double.NaN, Double.POSITIVE_INFINITY), 0.0);
+        Assert.assertEquals(-500, Util.min(500, -500), 0.0);
+        Assert.assertEquals(-500, Util.min(-500, 0), 0.0);
+        Assert.assertEquals(-1000, Util.min(-500, -1000), 0.0);
+    }
+
+    @Test
+    public void testParseBoolean() {
+        Assert.assertFalse(Util.parseBoolean("false"));
+        Assert.assertFalse(Util.parseBoolean("o== hh"));
+        Assert.assertFalse(Util.parseBoolean(null));
+
+        Assert.assertTrue(Util.parseBoolean("on"));
+        Assert.assertTrue(Util.parseBoolean("y"));
+        Assert.assertTrue(Util.parseBoolean("yes"));
+        Assert.assertTrue(Util.parseBoolean("1"));
+        Assert.assertTrue(Util.parseBoolean("true"));
+    }
+
+    @Test
+    public void testParseColor() {
+        // Test without alpha
+        final Color color1 = (Color) Util.parseColor("#AABBCC");
+        Assert.assertEquals(170, color1.getRed());
+        Assert.assertEquals(187, color1.getGreen());
+        Assert.assertEquals(204, color1.getBlue());
+        Assert.assertEquals(255, color1.getAlpha());
+
+        // Test with alpha
+        final Color color2 = (Color) Util.parseColor("#AABBCCDD");
+        Assert.assertEquals(170, color2.getRed());
+        Assert.assertEquals(187, color2.getGreen());
+        Assert.assertEquals(204, color2.getBlue());
+        Assert.assertEquals(221, color2.getAlpha());
+
+        // Test without hash
+        final Color color3 = (Color) Util.parseColor("AABBCCDD");
+        Assert.assertEquals(170, color3.getRed());
+        Assert.assertEquals(187, color3.getGreen());
+        Assert.assertEquals(204, color3.getBlue());
+        Assert.assertEquals(221, color3.getAlpha());
+
+        // Test invalid
+        thrown.expect(IllegalArgumentException.class);
+        Util.parseColor("not a  color string");
+    }
+
+    @Test
+    public void testSum() {
+        Assert.assertEquals(0.0, Util.sum(Double.NaN, 0), 0.0);
```

Comment:
I'd quite like to see this test a range of possible inputs (specifically negative values). It's not necessary, but it might make the PR just a little more robust.